6.CalculateOverdueFees.sql

```diff
@@ -0,0 +1,24 @@
+DROP FUNCTION IF EXISTS fn_CalculateOverdueFees;
+GO
+CREATE FUNCTION fn_CalculateOverdueFees(@LoanID INT)
+RETURNS int
+AS
+BEGIN
+    DECLARE @OverdueFee int;
+    DECLARE @DueDate DATE;
+    DECLARE @DateReturned DATE;
+	DECLARE @DaysOverdue INT;
+    SELECT @DueDate = DueDate, @DateReturned = DateReturned
+    FROM Loans
+    WHERE LoanID = @LoanID;
+    IF @DateReturned IS NULL
+        SET @DateReturned = GETDATE();
+    SET @DaysOverdue = DATEDIFF(DAY, @DueDate, @DateReturned);
+    IF @DaysOverdue <= 30
+        SET @OverdueFee = @DaysOverdue; 
+    ELSE
+        SET @OverdueFee = 30 + @DaysOverdue * 2.00; 
```

Comment:
We count the first 30 days as 1$ cost, but we want to eliminate them when calculating for over 30 days costs
Meaning: Fee = fee for 30 days +  (days overdue - 30) * cost for over 30 days
if you're late 31 days the fees will be 32$ and not 30 + 62 = 92$ 
```suggestion
        SET @OverdueFee = 30 + (@DaysOverdue - 30) * 2.00; 
```